### ISSUE TITLE
Annotations / hotspots

### ISSUE THREAD

**Issue by danieltome** [pisi/Reel#36]
Feature request.
The position will need to be based on each frame.

**Comment by pisi** [comment]
Ya, this one is a long-time wish list entry and it deserves its own ticket here, that's for sure.

I've been writing the hotspots module long ago, but unfortunatelly lost a good half of it in the process of backuping (how ironic) and that cut my steam a bit. It's actually rather easy to code (like you suggest it is just a matter of additional `frameChange` event handler, which would each time re-render the hotspots), it's just that my priorities lie elsewhere right now.

But it would definitely make a great 1.2 feature ;)

**Comment by danieltome** [comment]
Yeah. I will write this code, and commit it to my forked version. I'll let you know when I have it working.
If you have any half-baked code that could help out, let me know. I couldn't find any other branch. I'm quite new to github, so maybe I'm looking in the wrong place.

cheers,

**Comment by pisi** [comment]
Very cool! I welcome any kind of collaboration on jQuery Reel.

If you don't mind then, I've added you as Reel project collaborator here on Github and assigned you this ticket for the 1.2 milestone. Planning-wise speaking, 1.2 wouldn't be shipped till 2012.

Oh, and I will dig out the WIP code of mine, undust it and will get back to you soon to see if I can be of any assistance.

**Comment by danieltome** [comment]
Thanks!
I created a quick prototype with hotspots: http://danieltome.com/tests/reel/
At the moment it was just a test, I have to include this into the plugin somehow, and make easier to extend.
That would be my next goal. As well as creating a hotspot generator.
I had to add a trigger event called frameUpdate to jquery.reel.js, because the frameChange wasn't firing the framenumber.

**Comment by pisi** [comment]
Nice!

You are right, `frameChange` doesn't return a frame number. Actually, Reel stores all internal affairs on the respective DOM image node using jQuery `.data`. So instead of passing the value, I often rather favor retrieving of any value using the `.data`, against direct passing of just one or few arguments. 

For example the current frame can be determined simply by retrieving `.data("frame")`:

```
$reel.bind("frameChange", function(){
    $(this).data("frame");
});
```

Please, inspect `test/unit/data.js` tests for all other possible data keys one can read from Reel. Names are intended to be straightforward, but I will gladly explain whatever value isn't understood.

**Comment by pisi** [comment]
The annotations (formerly hotspots) feature is very close to finish. I've put together a little wiki page describing what annotations are and how they are used in Reel - [Annotations](/pisi/Reel/wiki/Annotations). Gentlemen, please if you could check it out. I would appreciate a feedback on the document if it's understandable enough. Thanks

**Comment by pisi** [comment]
I'm happy with the annotations code. Merged it into [development](/pisi/Reel/tree/development). I would still very much appreciate all testing and any feedback anyone can give me on it and [this](/pisi/Reel/issues/36#issuecomment-2883856).

Just some work on samplesto be included is needed and it should be done.

**Comment by pisi** [comment]
I finalized the annotations feature today and released [1.2alpha](/pisi/Reel/blob/v1.2alpha). Examples will come later in the beta (I didn't want to hold it any longer just because the examples aren't ready yet).

How annotations work is documented in the [Annotations](/pisi/Reel/wiki/Annotations) wiki article.
